### DISCUSSION TITLE
Playing with packets

### DISCUSSION THREAD

**Discussion by DeepVoyager111** [FabricMC/fabric-api#2553] [Mod Dev Support]
hello, i am a new fabric mod developer and i was wondering if anyone could tell me how to do the following:

- make an action in-game but not send the server a packet about it e.g. close inventory without packet
- cancel all packets to server for a few seconds (without getting disconnected)
- add buttons on screen when in gui

**Comment by itsmiir** [comment]
you can cancel whatever packets you want sent from the client. but a lot of servers expect those packets and will kick you if they don't receive them, or notice you doing things you shouldn't be able to based on the packets you've sent. the way to do that would be finding out where the packet is being sent and mixin to stop it. as for buttons, you can add them via mixin as well, by mixing into the respective Screen class.

**Reply by warjort** [reply]
You don't need mixins to modify screens.
You can use fabric-api's ScreenEvents, e.g. for what you doing AFTER_INIT
The javadoc in that class has more information.

**Reply by DeepVoyager111** [reply]
> you can cancel whatever packets you want sent from the client. but a lot of servers expect those packets and will kick you if they don't receive them, or notice you doing things you shouldn't be able to based on the packets you've sent. the way to do that would be finding out where the packet is being sent and mixin to stop it. as for buttons, you can add them via mixin as well, by mixing into the respective Screen class.

ok, but i need a code example. thank you.
